packages/catalog-api/src/lib/schema.graphql

```diff
@@ -8,27 +8,8 @@ type Query {
   """
   Queries custom elements in the entire catalog, from the latest version of each
   package.
-
-  Eventually this will have more query parameters, and use some sort of ranking
-  algorithm, otherwise the order will just be defined by the database
-  implementation.
-
-  NOT IMPLEMENTED
-  DO_NOT_LAUNCH
-  """
-  elements(distTag: String = "latest", limit: Int): [CustomElement!]!
-
-  """
-  Retrieves the custom element data for a single element.
-
-  NOT IMPLEMENTED
-  DO_NOT_LAUNCH
   """
-  element(
-    packageName: String!
-    elementName: String!
-    tag: String = "latest"
-  ): CustomElement
+  elements(query: String, limit: Int): [CustomElement!]!
```

Comment:
Do we perhaps also want a strict lookup like the `element` API we had before, or will we instead support search prefixes like `package:foo element:bar`?

Reply:
I think that the `package()` query gives us what we had with `element()` - and we can make some of the fields like `customElements` into queries if we want that.

I think this query should grow to add both search operators and possibly subqueries/fields that get you from element to package here.

One consideration is how to implement structured + text search. If we move to use a search service do we do both Firestore + search service queries and mix them? If a search service offer structured fields do we delegate all searches to it?